pkg/component/networking/nodelocaldns/nodelocaldns.go

```diff
@@ -557,3 +566,29 @@ func (n *nodeLocalDNS) SetClusterDNS(dns []string) {
 func (n *nodeLocalDNS) SetDNSServers(servers []string) {
 	n.values.DNSServers = servers
 }
+
+func (n *nodeLocalDNS) SetIPFamilies(ipfamilies []gardencorev1beta1.IPFamily) {
+	n.values.IPFamilies = ipfamilies
+}
+
+func (n *nodeLocalDNS) GetIPVSAddress() (ipvsAddress string) {
+	ipFamiliesSet := sets.New[gardencorev1beta1.IPFamily](n.values.IPFamilies...)
+	if ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv4) && !ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv6) {
+		ipvsAddress = nodelocaldnsconstants.IPVSAddress
+	}
+	if ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv6) && !ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv4) {
+		ipvsAddress = nodelocaldnsconstants.IPVSIPv6Address
+	}
+	return
+}
+
+func (n *nodeLocalDNS) GetHealthAddress() (healthAddress string) {
+	ipFamiliesSet := sets.New[gardencorev1beta1.IPFamily](n.values.IPFamilies...)
+	if ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv4) && !ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv6) {
+		healthAddress = nodelocaldnsconstants.IPVSAddress
+	}
+	if ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv6) && !ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv4) {
+		healthAddress = fmt.Sprintf("[%s]", nodelocaldnsconstants.IPVSIPv6Address)
+	}
+	return
```

Comment:
Why not simplify?

```suggestion
	return fmt.Sprintf("[%s]", n. GetIPVSAddress())
```

Reply:
The first part I would still need as I only want to add the brackets in the ipv6 case.

Reply:
Fair point. Maybe it is best to have a single function with a parameter whether to put brackets around IPv6 addresses. Then, you can have two simple functions reusing this helper function to have a clean interface.

Reply:
I prefer it this way, as it will already work for dual stack no matter which IP is in.